src/components/order-service/order-service.controller.ts

```diff
@@ -218,18 +220,29 @@ export class OrderServiceController implements angular.IController {
   };
 
   public createService() {
-    let serviceInstance = this.makeServiceInstance();
+    let parameters = this.getParameters();
+    let secretName: string = _.isEmpty(parameters) ? null : this.generateSecretName();
+    let serviceInstance = this.makeServiceInstance(secretName);
     let resource = {
       group: 'servicecatalog.k8s.io',
       resource: 'instances'
     };
     let context = {
       namespace: this.ctrl.selectedProject.metadata.name
     };
-    this.DataService.create(resource, null, serviceInstance, context).then((data: any) => {
+    this.DataService.create(resource, null, serviceInstance, context).then((serviceInstance: any) => {
       this.ctrl.orderInProgress = true;
-      this.watchResults(resource, data, context);
-      this.ctrl.serviceInstance = data;
+      this.watchResults(resource, serviceInstance, context);
+      this.ctrl.serviceInstance = serviceInstance;
+
+      // Create the parameters secret if necessary.
+      if (secretName) {
+        let secret = this.makeParametersSecret(secretName, parameters, serviceInstance);
+        this.DataService.create('secrets', null, secret, context).then(_.noop, (e: any) => {
+          this.ctrl.error = _.get(e, 'data');
+        });
```

Comment:
Is there an href to the secret that would get could get and pass to the bind results component now?

Reply:
There are two secrets now: one for the parameters and one for the bind result. I think the bind result is the interesting one. I'm not sure it's worth linking to the parameters secret.

Ideally we'd just have the instance/bind parameters on the provisioned service detail page.

Reply:
OK, gotcha.